src/simulation/deviceInterface/tempMeasurement/tempMeasurement.cpp

```diff
@@ -0,0 +1,143 @@
+    this->senBias = 0.0;
+    this->walkBounds = 1E-15; //don't allow random walk by default
+    this->noiseModel = GaussMarkov(1, this->RNGSeed);
+    this->faultState = TEMP_FAULT_NOMINAL;
+    this->pastValue = 0.0;
+    this->stuckValue = 0.0;
+    this->spikeProbability = 0.1; // default 10% chance of spiking if in fault state
+    this->spikeAmount = 2.0; // default to double the value if spiking
+}
+
+/*! Module Destructor */
+TempMeasurement::~TempMeasurement()
+{
+}
+
+/*! This method is used to reset the module and checks that required input messages are connected.
+    @return void
+*/
+void TempMeasurement::Reset(uint64_t CurrentSimNanos)
+{
+    // check that required input messages are connected
+    if (!this->tempInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "TempMeasurement.tempInMsg was not linked.");
+    }
+
+    if (this->spikeProbability > 1.0 || this->spikeProbability < 0.0) {
+        bskLogger.bskLog(BSK_ERROR, "The probability of temperature spike on fault must be between 0 and 1.");
+    }
+    // set up gaussMarkov and random number generator parameters
+    Eigen::VectorXd nMatrix;
+    Eigen::VectorXd pMatrix;
+    Eigen::VectorXd bounds;
+    nMatrix.resize(1,1);
+    pMatrix.resize(1,1);
+    bounds.resize(1,1);
+    
+    this->noiseModel.setRNGSeed(this->RNGSeed);
+    
+    nMatrix(0,0) = this->senNoiseStd;
+    this->noiseModel.setNoiseMatrix(nMatrix);
+    
+    bounds(0,0) = this->walkBounds;
+    this->noiseModel.setUpperBounds(bounds);
+    
+    pMatrix(0,0) = 1.;
+    this->noiseModel.setPropMatrix(pMatrix);
+
+    this->spikeProbabilityGenerator.seed((unsigned int)this->RNGSeed);
+
+}
+
+
+void TempMeasurement::applySensorErrors()
+{
+    // apply noise and bias
+    double sensorError;
+    if(this->senNoiseStd <= 0.0){
+        sensorError = this->senBias;
+    } else {
+        // get current error from gaussMarkov random number generator
+        this->noiseModel.computeNextState();
+        Eigen::VectorXd currentErrorEigen = this->noiseModel.getCurrentState();
+        double sensorNoise = currentErrorEigen.coeff(0,0);
+        sensorError = this->senBias + sensorNoise;
+    }
+    this->sensedTemperature = this->trueTemperature + sensorError;
+    
+    // apply fault conditions
+    if(this->faultState == TEMP_FAULT_STUCK_VALUE){ // stuck at specified value
+        this->sensedTemperature = this->stuckValue;
+    } else if (this->faultState == TEMP_FAULT_STUCK_CURRENT){ // stuck at last value before flag turned on
+        this->sensedTemperature = this->pastValue; 
+    } else if (this->faultState == TEMP_FAULT_SPIKING){ // spiking periodically with specified probability
+        // have to make a new distribution every time because SWIG can't parse putting this in the H file....?
+        std::uniform_real_distribution<double> spikeProbabilityDistribution(0.0,1.0); 
+        double n = spikeProbabilityDistribution(spikeProbabilityGenerator); // draw from uniform distribution
```

Comment:
:small_red_triangle_down: Missing the self reference, should be `this->spikeProbabilityGenerator` .

Reply:
Done.